blackbelt/commands/dep.py

```diff
@@ -17,7 +18,10 @@ def validate_dep(ctx, param, dep):
         raise click.BadParameter('The dependency format should be e.g. react@16.2')
     else:
         if dep_version == 'latest':
-            dep_version = run(['npm', 'view', dep_name, 'version'])
+            try:
+                dep_version = run(['npm', 'view', dep_name, 'version'])
+            except subprocess.CalledProcessError:
```

Comment:
I think it would be good to look into exit codes / status messages, because this wraps multiple errors under one hood and can make it very tricky to debug (i.e. if there is any other general npm or network problem).

Reply:
Perhaps it would be simpler to use the NPM web API directly, instead of invoking the CLI tool and trying to interpret it and it's response.

Can simply do a GET request to get the package information `https://api.npms.io/v2/package/dredd` and look for version in there. `404` response would indicate missing package.

Python one liner, can be broken into more for a `response.raise_for_status():

```python
>>> requests.get('https://api.npms.io/v2/package/dredd').json()['collected']['metadata']['version']
'4.9.3'
```

https://api-docs.npms.io

Reply:
@Almad Yup, my solution is a hotfix.
@kylef That's an amazing finding/an idea. I support you mentally in sending a PR! 😏